src/plugins/initiatives/initiativesDirectory.js

```diff
@@ -103,3 +106,47 @@ export function _initiativeAddress(
     ...NodeAddress.toParts(_trackerAddress(remoteUrl, fileName))
   );
 }
+
+// Checks the path exists and is a directory.
+// Returns the absolute path or throws.
+export async function _validatePath(localPath: string): Promise<string> {
+  const absPath = path.resolve(localPath);
+  if (!(await fs.exists(absPath))) {
+    throw new Error(
+      `Provided initiatives directory does not exist at: ${absPath}`
+    );
+  }
+  if (!(await fs.lstat(absPath)).isDirectory()) {
+    throw new Error(
+      `Provided initiatives directory is not a directory at: ${absPath}`
+    );
+  }
+  return absPath;
+}
+
+// Gets all *.json filenames in the given directory.
+export async function _findFiles(
+  localPath: string
+): Promise<$ReadOnlyArray<string>> {
+  const absoluteFileNames = await globby(path.join(localPath, "*.json"));
```

Comment:
globby supports `**` expansion. Why not do `**/*.json` and get subdirectory support from the start?

Reply:
For now this is less complex. As technically we would need to represent this as an array in the ID / NodeAddress and let the separator be handled by `path.join` and `[].join("/")` for URLs.

Reply:
I don't think we need to represent it as an array in the id / NodeAddress. We could just have the full subpath be a component, as in `["sourcecred", "initiative", "INITIATIVE_FILE", "https://github.com/sourcecred/cred/master/", "path/to/my/file.json"]`.

Reply:
And when your windows filesystem returns `path\to\my\file.json` instead? We're now responsible for normalizing filesystem paths and URLs :D

It can be done! But I'm just punting on it for now to make this a later improvement.

Reply:
Oh, I didn't realize that Windows has a different file path representation. Which is kind of embarrassing as I now see that I've [written bugs around backslash handling in the past](https://github.com/tensorflow/tensorboard/issues/51). Cool, let's revisit this later. :)